### DISCUSSION TITLE
Having trouble adding Windows workers

### DISCUSSION THREAD

**Discussion by tylerhowdcgm** [rancher/rke2#5158] [Q&A] [answered]
I currently have a cluster with three Linux server nodes and three Linux Worker nodes. I installed and specified calico as my CNI and calico is currently running as my CNI. Im currently running v1.26.11+rke2r1

```
NAME                                       READY   STATUS    RESTARTS   AGE
calico-kube-controllers-796658db69-98njm   1/1     Running   0          142m
calico-node-7twhp                          1/1     Running   0          55m
calico-node-b4mpd                          1/1     Running   0          137m
calico-node-d5586                          1/1     Running   0          58m
calico-node-fr8k6                          1/1     Running   0          142m
calico-node-nzsgh                          1/1     Running   0          140m
calico-node-rhmht                          1/1     Running   0          68m
calico-typha-db8b7ff8-sdtks                1/1     Running   0          142m
```

However when I try to start the rke2 service on my first Windows node I get this in event viewer.

```
The following information was included with the event: 

Unsupported CNI: canal
```

What did I do wrong?

**Comment by brandond** [comment]
Can you show how you've configured the Linux server nodes, and the configuration on the Windows nodes?

**Reply by tylerhowdcgm** [reply]
Sure Thing
Linux Servers
```
write-kubeconfig-mode: "0644"
token:
  - "<secret token>"
tls-san:
  - "kube-api.<internal.domain>"
node-label:
  - "os=linuxserver"
cni:
  - "calico"
```
Windows Workers
```
server: https://kube-api.<internal.domain>:9345
token: <matching secret token>
```
Im following this guide here: https://docs.rke2.io/install/quickstart#linux-agent-worker-node-installation

**Reply by brandond** [reply]
Are you sure that the `kube-api.<internal.domain>` address points ONLY at the servers using calico? I'm not sure how this would happen, other than pointing your windows nodes at cluster members that are not using the correct CNI.

@manuelbuil do you have any ideas?

**Reply by tylerhowdcgm** [reply]
Yes, im 100% sure. Im using an external haproxy to route all traffic and currently the api and supervisor are pointed at the server nodes. I also checked for running cni's with ps and calico is the only cni running on all Linux nodes. I did check the helm deployments in kube-system. There is a pod `helm-install-rke2-canal-6pw4b` , but it failed. Im not sure why it would attempt to deploy canal if I specify calico. Here are the logs from the canal helm pod.

```Error: INSTALLATION FAILED: rendered manifests contain a resource that already exists. Unable to continue with install: CustomResourceDefinition "bgpconfigurations.crd.projectcalico.org" in namespace "" exists and cannot be imported into the current release: invalid ownership metadata; annotation validation error: key "meta.helm.sh/release-name" must equal "rke2-canal": current value is "rke2-calico-crd"```

**Reply by brandond** [reply, marked as answer]
Can you confirm that you have `cni: calico` set on ALL of your servers?

Can you provide the output of `kubectl get node -o yaml | grep node-args` ?

**Reply by tylerhowdcgm** [reply]
I only see `"--cni","calico"` on my first server node. Could that be the issue?

**Reply by tylerhowdcgm** [reply]
I added the `cni: calico` to all my Linux nodes, then restarted the server/worker services and the Windows worker joined! Thank you so much for all your help!

**Reply by brandond** [reply]
That was what I was getting at... all the servers need to be configured identically. If there are any other settings that you forgot to sync up across the servers, now's a good time to check on that.

I am honestly surprised that you didn't run into more problems with this, I would have expected the linux nodes to be somewhat confused by this as well.
